mijia/src/introspect.rs

```diff
@@ -0,0 +1,215 @@
+    pub args: Vec<MethodArg>,
+    #[serde(rename = "annotation", default)]
+    pub annotations: Vec<Annotation>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct Signal {
+    pub name: String,
+    #[serde(rename = "arg", default)]
+    pub args: Vec<SignalArg>,
+    #[serde(rename = "annotation", default)]
+    pub annotations: Vec<Annotation>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct Property {
+    pub name: String,
+    #[serde(rename = "type")]
+    pub dbustype: String,
+    pub access: Access,
+    #[serde(rename = "annotation", default)]
+    pub annotations: Vec<Annotation>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct MethodArg {
+    pub name: Option<String>,
+    #[serde(rename = "type")]
+    pub dbustype: String,
+    #[serde(default = "default_method_arg_direction")]
+    pub direction: Direction,
+    #[serde(rename = "annotation", default)]
+    pub annotations: Vec<Annotation>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct SignalArg {
+    pub name: Option<String>,
+    #[serde(rename = "type")]
+    pub dbustype: String,
+    #[serde(default = "default_signal_arg_direction")]
+    pub direction: Direction,
+    #[serde(rename = "annotation", default)]
+    pub annotations: Vec<Annotation>,
+}
+
+fn default_method_arg_direction() -> Direction {
+    Direction::In
+}
+
+fn default_signal_arg_direction() -> Direction {
+    Direction::Out
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct Annotation {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, PartialEq, Eq)]
+pub enum Direction {
+    #[serde(rename = "in")]
+    In,
+    #[serde(rename = "out")]
+    Out,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, PartialEq, Eq)]
+pub enum Access {
+    #[serde(rename = "readwrite")]
+    ReadWrite,
+    #[serde(rename = "read")]
+    Read,
+    #[serde(rename = "write")]
+    Write,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn example_config() {
+        let value: Node = serde_xml_rs::from_str(
+            r#"<!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Object Introspection 
+        1.0//EN"
+        "http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+        <node><interface name="org.freedesktop.DBus.Introspectable"><method 
+        name="Introspect"><arg name="xml" type="s" direction="out"/>
+        </method></interface><interface 
+        name="org.freedesktop.DBus.ObjectManager"><method 
+        name="GetManagedObjects"><arg name="objects" type="a{oa{sa{sv}}}" 
+        direction="out"/>
+        </method><signal name="InterfacesAdded"><arg name="object" type="o"/>
+        <arg name="interfaces" type="a{sa{sv}}"/>
+        </signal>
+        <signal name="InterfacesRemoved"><arg name="object" type="o"/>
+        <arg name="interfaces" type="as"/>
+        </signal>
+        </interface><node name="org"/></node>"#,
```

Comment:
I know this is straight off the wire, but maybe dump this through `xmllint --format -` like we do in `bluez-generated/introspect.sh` (you could even `include_str!()` something in bluez-generated/specs/ if you were feeling fancy)

Reply:
Done.